system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs

```diff
@@ -1592,34 +1676,46 @@ impl_runtime_apis! {
 	impl xcm_runtime_apis::fees::XcmPaymentApi<Block> for Runtime {
 		fn query_acceptable_payment_assets(xcm_version: xcm::Version) -> Result<Vec<VersionedAssetId>, XcmPaymentApiError> {
 			let native_asset = xcm_config::KsmLocation::get();
+			let native_asset_v4 = xcm_config::KsmLocationV4::get();
 			// We accept the native asset to pay fees.
 			let mut acceptable_assets = vec![AssetId(native_asset.clone())];
 			// We also accept all assets in a pool with the native token.
-			acceptable_assets.extend(
-				assets_common::PoolAdapter::<Runtime>::get_assets_in_pool_with(native_asset)
-				.map_err(|()| XcmPaymentApiError::VersionedConversionFailed)?
-			);
+			let assets_in_pool_with_native = assets_common::get_assets_in_pool_with::<
+				Runtime,
+				xcm::v4::Location,
+			>(&native_asset_v4).map_err(|()| XcmPaymentApiError::VersionedConversionFailed)?;
+			acceptable_assets.extend(assets_in_pool_with_native);
+
 			PolkadotXcm::query_acceptable_payment_assets(xcm_version, acceptable_assets)
 		}
 
 		fn query_weight_to_asset_fee(weight: Weight, asset: VersionedAssetId) -> Result<u128, XcmPaymentApiError> {
-			let native_asset = xcm_config::KsmLocation::get();
+			let native_asset = KsmLocation::get();
+			let native_asset_v4 = KsmLocationV4::get();
 			let fee_in_native = WeightToFee::weight_to_fee(&weight);
-			let latest_asset_id: Result<AssetId, ()> = asset.clone().try_into();
-			match latest_asset_id {
-				Ok(asset_id) if asset_id.0 == native_asset => {
+			let v4_asset: Result<xcm::v4::AssetId, ()> = asset.clone().try_into();
+			match v4_asset {
+				Ok(asset_id) if asset_id.0 == native_asset_v4 => {
 					// for native asset
 					Ok(fee_in_native)
 				},
 				Ok(asset_id) => {
-					// Try to get current price of `asset_id` in `native_asset`.
-					if let Ok(Some(swapped_in_native)) = assets_common::PoolAdapter::<Runtime>::quote_price_tokens_for_exact_tokens(
```

Comment:
We should revert all these changes plus return back all those removed `xcm_payment_api_works` tests - this will reduce this PR also.

Relates to: https://github.com/paritytech/polkadot-sdk/pull/8174

Reply:
Quick question: Will that backport means we need to wait until `stable2412-5`?

Reply:
No, we can do patch release for just those two crates as soon as backport is merged

Reply:
This is not blocking this PR or this release, we will not wait for it, if it comes in time ok, otherwise it'll have to come as a patch release later.

Reply:
fixed = backported and reverted